internal/venice-common/src/main/java/com/linkedin/venice/controllerapi/D2ControllerClient.java

```diff
@@ -96,24 +114,27 @@ protected String discoverLeaderController() {
   }
 
   private static <RESPONSE> RESPONSE d2ClientGet(
-      D2Client d2Client,
+      List<D2Client> d2ClientList,
       String d2ServiceName,
       String path,
       QueryParams params,
       Class<RESPONSE> responseClass) {
     String requestPath = D2_SCHEME + d2ServiceName + path + "?" + encodeQueryParams(params);
-    try {
-      RestResponse response = D2ClientUtils.sendD2GetRequest(requestPath, d2Client);
-      String responseBody = response.getEntity().asString(StandardCharsets.UTF_8);
-      return ObjectMapperFactory.getInstance().readValue(responseBody, responseClass);
-    } catch (Exception e) {
-      throw new VeniceException("Failed to get response for url: " + requestPath + " with D2 Client", e);
+    for (D2Client d2Client: d2ClientList) {
+      try {
+        RestResponse response = D2ClientUtils.sendD2GetRequest(requestPath, d2Client);
+        String responseBody = response.getEntity().asString(StandardCharsets.UTF_8);
+        return ObjectMapperFactory.getInstance().readValue(responseBody, responseClass);
+      } catch (Exception e) {
+        LOGGER.error("Failed to get response for url: " + requestPath + " with D2 Client:" + d2Client, e);
+      }
     }
+    throw new VeniceException("List of D2 Clients failed to get response for url: " + requestPath);
   }
 
   public static D2ServiceDiscoveryResponse discoverCluster(D2Client d2Client, String d2ServiceName, String storeName) {
```

Comment:
`discoverCluster` will also need a method to support a list of D2 clients. A Leader controller is a property of a cluster, and the call to fetch the leader controller info needs to be preceded by a cluster discovery call. Only adding resiliency to the leader controller call is not sufficient to provide the resiliency